pkg/policy/distillery.go

```diff
@@ -118,7 +120,10 @@ func (cache *policyCache) updateSelectorPolicy(identity *identityPkg.Identity, e
 		return nil, false, err
 	}
 
-	cip.setPolicy(selPolicy, endpointID)
+	// There is now an outstanding reference to this policy, so increment the counter
+	selPolicy.L4Policy.acquire()
```

Comment:
I find it a bit confusing at first glance that its running `L4Policy.acquire()` here, and then immediately again inside of `cip.setPolicy`. After some thinking the logic looks sound, but its a bit hard to wrap your head around. I wonder if tis better to add this line just before the `return` instead, so that both the returns look roughly the same.